Remove Onboarding related Properties from LocalData (EXPOSUREAPP-5375)

kolyaopahle opened the pull request:
### Description

### Note that this PR does not provide any migration functionality from LocalData -> Settings, this will be done in a later PR

This PR moves all Onboarding related properties of LocalData into OnboardingData which is a injectable Singleton and uses the FlowPreference class to supply the values

kolyaopahle committed:
Removed Onboarding related Properties from LocalData
Added OnboardingData which provides the FlowPreferences that were defined in LocalData beforehand
Currently no migration for these values is implemented

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Turns out MockFlowPreference and MockSharedPreferences were not usable in the device test flavour
moved them into testShared

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

d4rken commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt:
```diff
@@ -23,89 +23,6 @@ object LocalData {
-     * @return boolean if user is onboarded
-     */
-    fun isOnboarded(): Boolean = getSharedPreferenceInstance().getBoolean(
-        CoronaWarnApplication.getAppContext().getString(R.string.preference_onboarding_completed),
```
Remove from strings.xml too?

d4rken commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingData.kt:
```diff
@@ -0,0 +1,48 @@
+import org.joda.time.Instant
+import javax.inject.Inject
+
+class OnboardingData @Inject constructor(
```
Where does the migration happen?

kolyaopahle replied:
Nowhere yet, i still want to align with @jurajkusner on the specifics of how to migrate

kolyaopahle replied:
Migration will be done for all preferences in a seperate PR

kolyaopahle committed:
OnboardingData -> OnboardingSettings
OnboardingSettings:
moved to get / set instead of FlowPreferences when subscribing is not needed
isOnboarded is now based on onboardingCompletedTimestamp != null instead of set seperately

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Removed Onboarding Preference key strings from all languages

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Merge branch 'feature/2850-refactor-localdata' into feature/5375-remove-onboarding-local-data

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt:
```diff
@@ -0,0 +1,38 @@
+
+    companion object {
+        private const val ONBOARDING_COMPLETED_TIMESTAMP = "onboarding.done.timestamp"
+        private const val BACKGROUND_CHECK_DONE = "onboarding.background.checked"
```
should not the keys match old keys? or a migration will implemented to copy over these data ?

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingActivity.kt:
```diff
@@ -71,9 +75,12 @@ class OnboardingActivity : AppCompatActivity(), LifecycleObserver, HasAndroidInj
+
+        settings.lastChangelogVersion.update {
+            BuildConfigWrap.VERSION_CODE
+        }
```
what about `isOnboarded `? it is removed ,but no saved using the new class

Kolpa replied:
`isOnboarded` has been replaced with a pure getter based on `onboardingCompletedTimestamp != null` to create a unified datasource and remove desync opportunities

mtwalli commented on Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/OnboardingSettings.kt:
```diff
@@ -0,0 +1,38 @@
+import org.joda.time.Instant
+import javax.inject.Inject
+
+class OnboardingSettings @Inject constructor(
```
this class is injected in Application ,is not better to annotate it with Singleton?

Kolpa replied:
whoops, missed that, absolutely

kolyaopahle committed:
Merge branch 'feature/2850-refactor-localdata' into feature/5375-remove-onboarding-local-data

kolyaopahle committed:
Added Singleton annotation to OnboardingSettings
Added migration code for OnboardingSettings to EncryptedPreferencesMigration

Signed-off-by: Kolya Opahle <k.opahle@sap.com>

kolyaopahle committed:
Merge branch 'feature/2850-refactor-localdata' into feature/5375-remove-onboarding-local-data

d4rken reviewed (approved):
lgtm

mtwalli reviewed (approved):
lgtm
